src/core/server/http/http_service.test.ts

```diff
@@ -248,6 +258,43 @@ test('register route handler', async () => {
   expect(registerRouterMock).toHaveBeenLastCalledWith(router);
 });
 
+test('returns `preboot` http server contract on preboot', async () => {
+  const configService = createConfigService();
+  const httpServer = {
+    server: fakeHapiServer,
+    registerStaticDir: jest.fn(),
+    auth: Symbol('auth'),
+    basePath: Symbol('basePath'),
+    csp: Symbol('csp'),
+  };
+
+  mockHttpServer.mockImplementation(() => ({
+    isListening: () => false,
+    setup: jest.fn().mockReturnValue(httpServer),
+    start: noop,
+    stop: noop,
+  }));
+
+  const service = new HttpService({ coreId, configService, env, logger });
+  await expect(service.preboot(prebootDeps)).resolves.toMatchInlineSnapshot(`
+          Object {
+            "auth": Symbol(auth),
+            "basePath": Symbol(basePath),
+            "csp": Symbol(csp),
+            "externalUrl": ExternalUrlConfig {
```

Comment:
Sure, will change. I usually prefer snapshots to validate the shape of the contracts since they will fail tests if you add new properties while `toMatchObject` will ignore them.